cmake_test/add_section.cmake

```diff
@@ -0,0 +1,41 @@
+include_guard()
+
+
+#[[[
+# Adds a test section, should be called inside of a declared test function directly before declaring the section function.
+# The NAME parameter will be populated as by set() with the generated section function name. Declare the section function using this generated name. Ex:
+#
+# .. code-block:: cmake
+#
+#    #This is inside of a declared test function
+#    ct_add_section(NAME this_section)
+#    function(${this_section})
+#        message(STATUS "This code will run in a test section")
+#    endfunction()
+#
+# :param EXPECTFAIL: Option indicating whether the section is expected to fail or not, if specified will cause the section to be ran in a subprocess.
+# :param NAME name: Required argument specifying the name variable of the section. Will set a variable with specified name containing the generated function ID to use.
+#]]
+macro(ct_add_section)
+
+    #TODO Set sections as a subproperty of CT_CURRENT_EXECUTION_UNIT instead of as a single global variable
+    set(options EXPECTFAIL)
+    set(oneValueArgs NAME)
+    set(multiValueArgs "")
+    cmake_parse_arguments(CT_ADD_SECTION "${options}" "${oneValueArgs}"
+                          "${multiValueArgs}" ${ARGN} )
+
+    #[[_ct_add_test_guts("${_at_test_name}")
+    #return()
+    #]]
+
+    string(RANDOM ALPHABET "abcdefghijklmnopqrstuvwxyz" "${CT_ADD_SECTION_NAME}") #Generate random section ID, using only alphabetical characters
+    get_property(curr_exec_unit GLOBAL PROPERTY "CT_CURRENT_EXECUTION_UNIT")
```

Comment:
does it make sense to use the existing CMakePP functionality for this (see [here](https://github.com/CMakePP/CMakePPCore/blob/master/cmake/cmakepp_core/utilities/global.cmake))?

Reply:
It does make sense and I am in the process of changing everything over to that. Should the random identifier creation remain the same or is there something in CMakePP that would work better?

Reply:
We have [unique_id](https://github.com/CMakePP/CMakePPCore/blob/master/cmake/cmakepp_core/utilities/unique_id.cmake) which I think does the same thing.

Reply:
I have switched everything over to CMakePP globals and unique_id